src/sentry/api/bases/organization.py

```diff
@@ -381,27 +383,36 @@ def _filter_projects_by_permissions(
         include_all_accessible: bool = False,
     ) -> list[Project]:
         user = getattr(request, "user", None)
-        filtered = projects
+
         with sentry_sdk.start_span(op="apply_project_permissions") as span:
             span.set_data("Project Count", len(projects))
             if force_global_perms:
                 span.set_tag("mode", "force_global_perms")
-            else:
-                if (
-                    user
-                    and is_active_superuser(request)  # superuser should fetch all projects
-                    or not filter_by_membership  # explicitly requested projects
-                    or include_all_accessible  # requested $all projects
-                ):
-                    span.set_tag("mode", "has_project_access")
-                    func = request.access.has_project_access
+                return projects
+
+            active_superuser = user and is_active_superuser(request)
+            # Superuser should fetch all projects.
+            # Also check if explicitly requesting all projects with include_all_accessible
+            if active_superuser or include_all_accessible:
+                span.set_tag("mode", "has_project_access")
+                proj_filter = request.access.has_project_access
+            # Check if explicitly requesting specific projects
+            elif not filter_by_membership:
+                if user and is_active_staff(request):
+                    # There is a special case for staff, where we want to fetch usage stats for a single
+                    # project in _admin using OrganizationStatsEndpointV2 but cannot use has_project_access
+                    # like superuser because it fails. The workaround is to create a lambda that mimics
+                    # checking for active projects like has_project_access without further validation.
+                    span.set_tag("mode", "staff_fetch_all")
+                    proj_filter = lambda proj: proj.status == ObjectStatus.ACTIVE  # noqa: E731
                 else:
-                    span.set_tag("mode", "has_project_membership")
-                    func = request.access.has_project_membership
-
-                filtered = [p for p in projects if func(p)]
+                    span.set_tag("mode", "has_project_access")
+                    proj_filter = request.access.has_project_access
```

Comment:
nit: would be awesome if we could fit this into the first if with active_superuser since it's the same logic setting

Reply:
I thought out that originally but we actually need to separate the logic for superuser and staff here. If combine `or not filter_by_membership`, this would actually be true for the staff case so the staff check would have to get moved into the single big if.

There could be a case where you have both modes active but are browsing SAAS, in which case you'd enter the superuser check but use the staff lambda instead.